### DISCUSSION TITLE
Unable to upload attachments to document

### DISCUSSION THREAD

**Discussion by ryanrank** [apache/couchdb#4237] [Q&A]
I am trying to upload attachments to documents in my DB. A tiny attachment works, for instance a text file of just a few bytes. But a JPG that is 124k fails. If I upload it using javascript, I get a `net::ERR_EMPTY_RESPONSE` in the console. If I upload it in the Web UI, I get `Error uploading file` If I look at the logs, I don't see an indication of the attempted upload. Admittedly, I'm not as familiar with CouchDB as I'd like to be, so it's possible that I'm overlooking the log.

I'm guessing there is some setting(s) in the local.ini file that I need to set which will help with the write of the file upload. I'm just not sure which setting(s). Any assistance would be useful.

Version: 3.2.2
Running on a Kubernetes setup with CouchDB exposed via NodePort
Using the official docker container, no custom configurations applied.

**Comment by big-r81** [comment]
Hi,

can you try your upload directly to CouchDB with `curl` and verify it uploads the file?

For example:

```bash
# create attachment db
curl -u user:password -X PUT http://127.0.0.1:15984/attdb

# create and upload document with attachment
curl -u user:password -X PUT http://127.0.0.1:15984/attdb/doc1/image1 --data-binary @path/to/image/image.jpg -H "Content-Type:image/jpg"
```

Is there anything recorded in your logfile?

**Reply by ryanrank** [reply]
This is what I'm seeing:
```
$ ls -l oldhim.jpg 
-rw-rw-r-- 1 owner group 234019 Oct 24 08:03 oldhim.jpg
$ curl -X PUT http://couchdb.domain.com:5984/db/uid13c7fb648d9d7f46800b08b84d013429uid/image.jpg?rev=14-f9caf5f084da3762b513aac7f3d29d35 --data-binary @oldhim.jpg -H "Content-Type:image/jpg"
curl: (52) Empty reply from server
$ echo "This is a test string" > text.txt
$ curl -X PUT http://couchdb.domain.com:5984/db/uid13c7fb648d9d7f46800b08b84d013429uid/text.txt?rev=14-f9caf5f084da3762b513aac7f3d29d35 --data-binary @text.txt -H "Content-Type:text/plain"
{"ok":true,"id":"uid13c7fb648d9d7f46800b08b84d013429uid","rev":"15-0d1b19b52999090de5a1d4879e92bddf"}
```
This is what I saw in the logs in the logs around the times of the uploads. No mention directly of the image upload, but you can see the text file uploaded as indicated but the curl output.
```
[error] 2022-10-25T01:36:57.956353Z nonode@nohost <0.26633.65> 0904afea78 fabric_worker_timeout update_docs,nonode@nohost,<<"shards/40000000-5fffffff/fab.1664496730">>
[error] 2022-10-25T01:38:37.461212Z nonode@nohost <0.27516.65> 5a28286bc8 fabric_worker_timeout update_docs,nonode@nohost,<<"shards/40000000-5fffffff/fab.1664496730">>
[notice] 2022-10-25T01:38:50.534437Z nonode@nohost <0.28105.65> d42ed8d198 couchdb.rank.family:5984 10.42.0.0 undefined PUT /fab/uid13c7fb648d9d7f46800b08b84d013429uid/text.txt?rev=14-f9caf5f084da3762b513aac7f3d29d35 201 ok 34
```
After a bunch of experimentation, it looks like the attachment breaks when the file is 10250 bytes. See below:
```
$ ls -l text.txt 
-rw-rw-r-- 1 owner group 10249 Oct 24 21:54 text.txt
$ curl -X PUT http://couchdb.domain.com:5984/db/uid13c7fb648d9d7f46800b08b84d013429uid/text.txt?rev=34-28d109520951f860abd05ba6ad70c1f1 --data-binary @text.txt -H "Content-Type:text/plain"
{"ok":true,"id":"uid13c7fb648d9d7f46800b08b84d013429uid","rev":"35-e80d52d1091fdee444bc9e533ae15120"}
$ ls -l text.txt 
-rw-rw-r-- 1 owner group 10250 Oct 24 21:57 text.txt
$ curl -X PUT http://couchdb.domain.com:5984/db/uid13c7fb648d9d7f46800b08b84d013429uid/text.txt?rev=35-e80d52d1091fdee444bc9e533ae15120 --data-binary @text.txt -H "Content-Type:text/plain"
curl: (52) Empty reply from server
```

**Reply by big-r81** [reply]
Is there a proxy in between?

**Reply by ryanrank** [reply]
I did more troubleshooting. It looks like there's a network configuration issue at the OS level of the host machine. I am unable to scp a 10k file or larger to the host that the container is running on. So this is *not* couchdb related.

Direct answer to your question: no proxy. However, considering the scp problem above, I'll consider this issue closed from the CouchDB perspective. I'll do some more research and troubleshooting, and if needed, ask help in a more appropriate forum. Thank you!
